### PR TITLE
Search: Include panel_type in query

### PR DESCRIPTION
**Pull request by ryantxu**
When featureFlag `panelTitleSearch` exists, we can also search by panel_type or datasource_uid -- the panel_type option did not survive a recent cleanup refactor, so this adds it back.

http://localhost:3000/dashboards?panel_type=timeseries
image

All dashboards using `gdev-testdata` http://localhost:3000/dashboards?datasource=PD8C576611E62080A
image

Here is a dashboard that lets you see panel_type and datasource usage in your instance:
image

<details><summary>Dashboard with search links</summary>

```json
{
  "annotations": {
    "list": [
      {
        "builtIn": 1,
        "datasource": {
          "type": "grafana",
          "uid": "-- Grafana --"
        },
        "enable": true,
        "hide": true,
        "iconColor": "rgba(0, 211, 255, 1)",
        "name": "Annotations & Alerts",
        "type": "dashboard"
      }
    ]
  },
  "editable": true,
  "fiscalYearStartMonth": 0,
  "graphTooltip": 0,
  "id": 199,
  "links": [],
  "liveNow": false,
  "panels": [
    {
      "datasource": {
        "type": "datasource",
        "uid": "grafana"
      },
      "fieldConfig": {
        "defaults": {
          "color": {
            "mode": "thresholds"
          },
          "custom": {
            "align": "auto",
            "cellOptions": {
              "type": "auto"
            },
            "inspect": false
          },
          "links": [
            {
              "title": "Search",
              "url": "/dashboards?panel_type=${__data.fields[0]}"
            }
          ],
          "mappings": [],
          "thresholds": {
            "mode": "absolute",
            "steps": [
              {
                "color": "green",
                "value": null
              },
              {
                "color": "red",
                "value": 80
              }
            ]
          }
        },
        "overrides": []
      },
      "gridPos": {
        "h": 10,
        "w": 9,
        "x": 0,
        "y": 0
      },
      "id": 2,
      "options": {
        "footer": {
          "countRows": false,
          "fields": "",
          "reducer": [
            "sum"
          ],
          "show": false
        },
        "frameIndex": 1,
        "showHeader": true,
        "showRowNums": false
      },
      "pluginVersion": "9.4.0-pre",
      "targets": [
        {
          "datasource": {
            "type": "datasource",
            "uid": "grafana"
          },
          "queryType": "search",
          "refId": "A",
          "search": {
            "ds_uid": "",
            "explain": false,
            "facet": [
              {
                "field": "panel_type"
              }
            ],
            "from": 0,
            "kind": [
              "panel"
            ],
            "limit": 1,
            "location": "",
            "query": "*",
            "sort": "",
            "tags": []
          }
        }
      ],
      "title": "Panel usage across instance",
      "type": "table"
    },
    {
      "datasource": {
        "type": "datasource",
        "uid": "grafana"
      },
      "fieldConfig": {
        "defaults": {
          "color": {
            "mode": "thresholds"
          },
          "custom": {
            "align": "auto",
            "cellOptions": {
              "type": "auto"
            },
            "inspect": false
          },
          "links": [
            {
              "title": "",
              "url": "/dashboards?datasource=${__data.fields[0]}"
            }
          ],
          "mappings": [],
          "thresholds": {
            "mode": "absolute",
            "steps": [
              {
                "color": "green",
                "value": null
              },
              {
                "color": "red",
                "value": 80
              }
            ]
          }
        },
        "overrides": []
      },
      "gridPos": {
        "h": 10,
        "w": 8,
        "x": 9,
        "y": 0
      },
      "id": 3,
      "options": {
        "footer": {
          "countRows": false,
          "fields": "",
          "reducer": [
            "sum"
          ],
          "show": false
        },
        "frameIndex": 1,
        "showHeader": true,
        "showRowNums": false
      },
      "pluginVersion": "9.4.0-pre",
      "targets": [
        {
          "datasource": {
            "type": "datasource",
            "uid": "grafana"
          },
          "queryType": "search",
          "refId": "A",
          "search": {
            "ds_uid": "",
            "explain": false,
            "facet": [
              {
                "field": "ds_uid"
              }
            ],
            "from": 0,
            "kind": [
              "dashboard"
            ],
            "limit": 1,
            "location": "",
            "query": "*",
            "sort": "",
            "tags": []
          }
        }
      ],
      "title": "Datasource instance usage in dashboards",
      "type": "table"
    },
    {
      "datasource": {
        "type": "datasource",
        "uid": "grafana"
      },
      "fieldConfig": {
        "defaults": {
          "color": {
            "mode": "thresholds"
          },
          "custom": {
            "align": "auto",
            "cellOptions": {
              "type": "auto"
            },
            "inspect": false
          },
          "mappings": [],
          "thresholds": {
            "mode": "absolute",
            "steps": [
              {
                "color": "green",
                "value": null
              },
              {
                "color": "red",
                "value": 80
              }
            ]
          }
        },
        "overrides": []
      },
      "gridPos": {
        "h": 10,
        "w": 7,
        "x": 17,
        "y": 0
      },
      "id": 4,
      "options": {
        "footer": {
          "countRows": false,
          "fields": "",
          "reducer": [
            "sum"
          ],
          "show": false
        },
        "frameIndex": 1,
        "showHeader": true,
        "showRowNums": false
      },
      "pluginVersion": "9.4.0-pre",
      "targets": [
        {
          "datasource": {
            "type": "datasource",
            "uid": "grafana"
          },
          "queryType": "search",
          "refId": "A",
          "search": {
            "ds_uid": "",
            "explain": false,
            "facet": [
              {
                "field": "ds_type"
              }
            ],
            "from": 0,
            "kind": [
              "dashboard"
            ],
            "limit": 1,
            "location": "",
            "query": "*",
            "sort": "",
            "tags": []
          }
        }
      ],
      "title": "Datasource type usage in dashboards",
      "type": "table"
    }
  ],
  "refresh": "",
  "revision": 1,
  "schemaVersion": 38,
  "style": "dark",
  "tags": [],
  "templating": {
    "list": []
  },
  "time": {
    "from": "now-6h",
    "to": "now"
  },
  "timepicker": {},
  "timezone": "",
  "title": "Usage",
  "uid": "f4a7d36b-5503-4d2d-a3e3-b13c83d26993",
  "version": 4,
  "weekStart": ""
}
```
</details>

### CODE REVIEW

File: public/app/features/search/page/components/columns.tsx [hunk 1]
```diff
@@ -170,7 +170,10 @@ export const generateColumns = (
         return (
           <div {...p.cellProps} className={cx(styles.locationCellStyle)}>
             {parts.map((p) => {
-              const info = meta.locationInfo[p];
+              let info = meta.locationInfo[p];
+              if (!info && p === 'general') {
+                info = { kind: 'folder', url: '/dashboards', name: 'General' };
```
Comment (ryantxu) [inline public/app/features/search/page/components/columns.tsx:175]: Before:
image

After:
image

**Review by toddtreece — COMMENTED** [review]
[this](http://localhost:3000/dashboards?panel_type=a) still returns results in the folder view:
image

**Comment by ryantxu** [comment]
@joshhunt @toddtreece can you take a look?  This is super helpful for tracking down migration issues with old panels on ops!

**Review by toddtreece — APPROVED** [review]
i can no longer switch to folder view, so it fixes the issue i was seeing 👍

**Pull request merged**